### ISSUE TITLE
Any plans to support reccuring events with rrule?

### ISSUE THREAD

**Issue by treighmawaka** [vkurko/calendar#28]
First off, great job putting this together! I've been looking for a viable alternative to fullcalendar and this is it. 

Can you please advise or point to how to handle recurring events? Fullcalendar has `rrule-plugin`. Do you offer anything similar? I couldn't find it in the docs.

Cheers!

**Comment by vkurko** [comment]
Hi @treighmawaka 

Thank you for the star! 🙂

As for the recurring events, unfortunately, I have no plans to support them at this time. But I'll leave this issue open and if more people require this feature then I will do it.

**Comment by treighmawaka** [comment]
@vkurko thanks for the quick reply.  I'll keep on eye on this, hopefully it's upvoted.

**Comment by meisemann1999** [comment]
I agree that this feature would be really helpful. I'm currently trying to find a workaround as my use case will need to support recurring events.

Thanks for this great project!

**Comment by amoscicki** [comment]
> we too are looking for this, if it’s something I can help with I’d be more than happy to, but svelte is new to me. If anyone has any workarounds or such that would be nice to share. Looking forward to contributing where I can!

I will be implementing recurrence in one of the solutions I am using this library, and I think present capabilities of the calendar are perfectly enough to achieve this. 

What I mean by that is since the calendar is only really meant to present the data in a visual way and responsibility of managing the data is not really something that should fall on the calendar itself. 
While viewing it the absolute max you should have inside events prop at any given time should be the amount that calendar can draw at a time. 
Think about it like having a box the size of the currently used view. 
If it's a week, you should have week's worth of data or you may easily run into performance challenges.

Instead you should have a layer of methods that dynamically select the data you want to be presented based on the currently displayed date range and if you want to indicate that event is recurring you can utilize html content of particular event by for example displaying some icon. 

Having that said I'm building some utilities that will support this project and example implementation of some templates that can be used as event content including recurring events could be in scope for that.

**Comment by steveb85** [comment]
@amoscicki  i agree 100% in what you said, that would be really helpful if you’re willing to share your outcomes too. The calendar is a nice representation of the data, but generating the data should happen outside that element.

**Comment by charm-sign** [comment]
> 你好@treighmawaka
> 
> 谢谢你的星星！🙂
> 
> 至于重复事件，很遗憾，我目前没有计划支持它们。但我将保留这个问题，如果更多人需要此功能，那么我会实现它。

你好，这个重复规则确实很实用，现在能够支持吗？

**Comment by vkurko** [comment]
> 你好，这个重复规则确实很实用，现在能够支持吗？

At the moment I would like to devote time to other tasks that in my opinion are of higher priority. So if someone wants to try to implement this functionality themselves, I will review their pull request.

**Comment by leninnava** [comment]
I actually pulled this out with some heavy parsing of my events source on top of it, plus using the rrule standard. I'll clone the repo so I can dig around and see if I can get a working solution.

**Comment by 66mya** [comment]
@leninnava Did you ever get this working?  We are looking for a tool that our staff can easily use to schedule educational programs under different resources and some of the programs are recurring.  This is completely internal, so I'm just trying to find something that works easily for our team.
